src/ImpactEvaluator.sol

```diff
@@ -33,8 +35,12 @@ contract ImpactEvaluator is AccessControl {
     function advanceRound() private {
         Round memory round;
         round.end = block.number + nextRoundLength;
+        round.exists = true;
         rounds.push(round);
         emit RoundStart(currentRoundIndex());
+        if (rounds.length > maxStoredRounds) {
+            delete rounds[rounds.length - maxStoredRounds - 1];
```

Comment:
The rounds array still grows, however we reduce cost by turning old rounds into empty elements

Reply:
This way we currently don't need to maintain a current round index either

Reply:
Next: Work on a data structure that won't grow above the threshold, while still staying in order, and not needing to copy all elements on purge

Reply:
(ring using linked lists)

Reply:
Resources:
- https://ethereum.stackexchange.com/questions/44392/recursive-struct-in-solidity
- https://programtheblockchain.com/posts/2018/03/30/storage-patterns-doubly-linked-list/